rules/integrations/azure/credential_access_entra_signin_brute_force_microsoft_365.toml

```diff
@@ -0,0 +1,82 @@
+[metadata]
+creation_date = "2024/09/06"
+integration = ["azure"]
+maturity = "production"
+min_stack_comments = "ES|QL not available until 8.13.0 in technical preview."
+min_stack_version = "8.13.0"
+updated_date = "2024/09/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies potential brute-force attempts against Microsoft 365 user accounts by detecting a high number of failed
+interactive or non-interactive login attempts within a 30-minute window. Attackers may attempt to brute force user
+accounts to gain unauthorized access to Microsoft 365 services via different services such as Exchange, SharePoint, or
+Teams.
+"""
+false_positives = [
+    """
+    Automated processes that attempt to authenticate using expired credentials or have misconfigured authentication
+    settings may lead to false positives.
+    """,
+]
+from = "now-15m"
+language = "esql"
+license = "Elastic License v2"
+name = "Azure Entra Sign-in Brute Force against Microsoft 365 Accounts"
+note = "This rule relies on Azure Entra ID sign-in logs, but filters for Microsoft 365 resources."
+references = [
+    "https://cloud.hacktricks.xyz/pentesting-cloud/azure-security/az-unauthenticated-enum-and-initial-entry/az-password-spraying",
+    "https://github.com/0xZDH/o365spray"
+]
+risk_score = 47
+rule_id = "35ab3cfa-6c67-11ef-ab4d-f661ea17fbcc"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: SaaS",
+    "Data Source: Azure",
+    "Data Source: Entra ID",
+    "Data Source: Entra ID Sign-in",
+    "Use Case: Identity and Access Audit",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-azure.signinlogs*
+// truncate the timestamp to a 30-minute window
+| eval target_time_window = DATE_TRUNC(30 minutes, @timestamp)
+| WHERE
+  event.dataset == "azure.signinlogs"
+  and event.category == "authentication"
+  and to_lower(azure.signinlogs.properties.resource_display_name) rlike "(.*)365(.*)"
+  and azure.signinlogs.category in ("NonInteractiveUserSignInLogs", "SignInLogs")
+  and event.outcome != "success"
+  // for tuning review azure.signinlogs.properties.status.error_code
+  // https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes
+// count the number of login sources and failed login attempts
+| stats
+  login_source_count = count(source.ip),
+  failed_login_count = count(*) by target_time_window, azure.signinlogs.properties.user_principal_name
```

Comment:
this rule covers the bruteforce targeting one user account, what about a new rule same source.ip targeting many user accounts (join by source.ip only or ASN)

Reply:
Yea for sure I'll add one for that as well!